File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        for(int i= 0; i<nums.length; i++){
            int check = nums[i];
            for(int j = 0; j<nums.length; j++){
                if(i == j){
                    break;
                }
                if(check == nums[j]){
                    return true;
                }
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        # Speichert bereits gesehene Zahlen und deren Indizes
        seen = {}
        
        for i, num in enumerate(nums):
            complement = target - num
            
            # Prüfen, ob das Gegenstück bereits im Dictionary existiert
            if complement in seen:
                # Gibt den kleineren Index zuerst zurück
                return [seen[complement], i]
            
            # Aktuelle Zahl mit ihrem Index speichern
            seen[num] = i

        